### ISSUE TITLE
Getting started with viewer.js

### ISSUE THREAD

**Issue by J1bz** [flowtsohg/mdx-m3-viewer#16]
Hi,

I've just cloned your repo on my work station but I'm not able to get started with a trivial example. Commands I ran:

 ```
 git clone https://github.com/flowtsohg/mdx-m3-viewer/
  cd mdx-m3-viewer
  ruby compiler.rb
  Generating the viewer
  compiler.rb:187:in `read': No such file or directory - src/spatial.js (Errno::ENOENT)
          from compiler.rb:187:in `block (2 levels) in compile_source'
          from compiler.rb:183:in `open'
          from compiler.rb:183:in `block in compile_source'
          from compiler.rb:179:in `each'
          from compiler.rb:179:in `compile_source'
          from compiler.rb:278:in `compile'
          from compiler.rb:281:in `<main>'
```

Although compiler.rb didn't manage to generate viewer.js, it generated some files in a build directory, build/spatial.js being one of them. I copied enough js files from build to src so that I was able to run compiler.rb with no error.

I guess build/output/viewer.js is the file to be included in a project. So I copied it in examples directory and opened examples/index.html in a browser. But the javascript console output those lines:

```
  ReferenceError: EventDispatcher is not defined viewer.js:5333:1
  ReferenceError: BaseModel is not defined objmodel.js:5:1
  ReferenceError: BaseModelInstance is not defined objmodelinstance.js:5:1
```

Since I was already working on assumptions so far, I'm seeking confirmations about how the build process is supposed to be done.

Maybe I've missed some steps, but it would be very helpful to have a detailed step-by-step procedure to get started with your example :)

Thanks

**Comment by flowtsohg** [comment]
I changed things and never quite kept the examples updated (nor are the docs).
About the compiler, I am not sure. I stopped using it, and it will be gone soon (perhaps in favor of a more standardized method, like babel, and a Python server for the examples??).
The shader compiling will, too, be gone, with the shaders stored in simple template (ES6) strings.

If you just want to test the built-in handlers, I could perhaps open a new repo, or update the w3x-w3m one, with the current code.
It's nowhere near finished, but it will give you a general idea of how the code is going to work in the future.
